client/src/pages/dashboard.tsx

```diff
@@ -0,0 +1,80 @@
+import { format, fromUnixTime, getUnixTime } from 'date-fns';
+import { DashboardLayout } from '../components/DashboardLayout';
+import { Overview } from '../components/Overview';
+import { DashboardData, PriceResult } from '../types/FileData';
+import { Container } from '../ui/Container';
+
+interface DashboardProps {
+  statsData: DashboardData;
+}
+
+const Dashboard = ({ statsData }: DashboardProps) => {
+  return (
+    <Container
+      css={{
+        display: 'flex',
+        flexDirection: 'column',
+        gap: '$5',
+        py: '$5',
+
+        '@lg': {
+          gap: '$20',
+          flexDirection: 'row',
+          py: '$10',
+          height: '100vh',
+        },
+
+        '@xl': {
+          gap: '128px',
+        },
+      }}
+    >
+      <DashboardLayout />
+      <Overview statsData={statsData} />
+    </Container>
+  );
+};
+
+export async function getStaticProps() {
+  const res = await fetch(process.env.DASHBOARD_API_URL!);
+  const stxStats = await res.json();
+
+  const priceRes = await fetch(process.env.STX_PRICE_API_URL!);
+  const priceData = await priceRes.json();
+
+  const today = new Date();
+  const fromDate = getUnixTime(
+    new Date(new Date().setDate(today.getDate() - 30))
+  );
+  const toDate = getUnixTime(Date.now());
+
+  const rangeRes = await fetch(
+    `${process.env.RANGE_API_URL}from=${fromDate}&to=${toDate}`
+  );
+  const rangeData = await rangeRes.json();
+  const priceEvolutionRes: PriceResult[] = rangeData.prices;
+
+  let stxPriceEvolution: {}[] = [];
+
+  priceEvolutionRes.forEach((element) => {
+    const formattedDate = format(fromUnixTime(element[0] / 1000), 'yyyy-MM-dd');
+
+    const target = { date: formattedDate, value: element[1].toFixed(1) };
+
+    const toObj = Object.assign({}, target);
```

Comment:
Is this needed? Can you please explain the logic here? :)

Reply:
Sure, this approach is the result of an issue I was having while trying to get the original data coming from the API into the AreaChart `data` prop. So the only way I managed to solve this was to transform the data structure to match the `StatsData[]` type of the data prop. Happy to try a cleaner/better approach if you have any idea of another solution :)